CHANGELOG.md

```diff
@@ -34,10 +24,31 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Misc
 
 ### Internal
-- Added an Internal section to the changelog. This section is intended to capture any notable non-public changes to the project.
+
 
 # Past Releases
 
+## [0.50.0]
+
+### Fixed
+
+- The `Environment` will now automatically inherit for `BlueprintView` instances nested inside `UIViewElement` during measurement.
+
+### Added
+
+- Introduced `ElementContent.init(byMeasuring:)`, for use when your `Element` contains a nested `BlueprintView`, commonly used to implement stateful elements. This avoids detached measurements and improves performance. 
+
+### Changed
+
+- Renamed `BlueprintViewUpdateMetrics` to `BlueprintViewRenderMetrics`.
+- Renamed `BlueprintViewRenderMetrics.measureDuration` to `layoutDuration`.
+- Renamed `BlueprintViewMetricsDelegate.blueprintView(_:completedUpdateWith:)` to `blueprintView(_:completedRenderWith:)`.
+- `BlueprintViewRenderMetrics` values are now calculated using `CACurrentMediaTime` instead of `Date`.
+- `ElementContent.init(child:)` now utilizes a unique `ContentStorage` type, which improves layout performance.
```

Comment:
this sort of thing should be in that new Internal section

Reply:
Yeah I wasn't 100% sure, since it in theory is a "change" of improved performance. I'll move it.